component/kafka/kafka.go

```diff
@@ -108,3 +108,18 @@ func DefaultConsumerSaramaConfig(name string, readCommitted bool) (*sarama.Confi
 
 	return config, nil
 }
+
+// DeduplicateMessages takes a slice of Messages and de-duplicates the messages based on the Key of those messages.
```

Comment:
Do we have guarantees that the messages in the slice are sorted by the Kafka metadata timestamp? e.g. from earliest to latest

And is that guarantee honored regardless of the way we consume the topic?

I think that once we know the answer to these we should specify it in the GoDoc

Reply:
Great questions :)

This implementation assumes messages are ordered, which is the default behaviour from Kafka unless the producer decided to change the partition hashing logic to something nondeterministic.
Kafka by default hashes the key and uses the hash for partition selection, so the same key will always go to the same partition. Kafka also guarantees ordering within partitions, so by default you get the guarantees, unless the producer decided otherwise.

If you meant are messages stored `[key: old, key: newer, key: newest]` in Patron, yes. The slice of messages is appended only (here: https://github.com/beatlabs/patron/blob/77223c31d98af664e10f18adf83e8c72c363b87c/component/kafka/group/component.go#L423)

I've added some additional documentation on this, do you think that covers it?

Reply:
Looks good :) I was just thinking that user of the API might wonder about this, so the GoDoc now will point them in the right direction.